src/ontodia/diagram/linkLayer.tsx

```diff
@@ -125,11 +126,50 @@ export class LinkLayer extends Component<LinkLayerProps, {}> {
         this.routings = this.router.route(this.props.view.model);
     }
 
+    private getNestedGroups = (group: string, groups: {[id: string]: string}): {[id: string]: string} => {
+        const {view} = this.props;
+        const elements = view.model.elements.filter(el => el.group === group);
```

Comment:
It's possible to combine `.filter` and `.forEach` here to avoid allocating memory for the `elements` array.